openpype/hosts/traypublisher/plugins/publish/validate_frame_ranges.py

```diff
@@ -0,0 +1,72 @@
+import re
+
+import pyblish.api
+
+import openpype.api
+from openpype import lib
+from openpype.pipeline import (
+    PublishXmlValidationError,
+    OptionalPyblishPluginMixin
+)
+
+
+class ValidateFrameRange(OptionalPyblishPluginMixin,
+                         pyblish.api.InstancePlugin):
+    """Validating frame range of rendered files against state in DB."""
+
+    label = "Validate Frame Range"
+    hosts = ["traypublisher"]
+    families = ["render"]
+    order = openpype.api.ValidateContentsOrder
+
+    optional = True
+    # published data might be sequence (.mov, .mp4) in that counting files
+    # doesnt make sense
+    check_extensions = ["exr", "dpx", "jpg", "jpeg", "png", "tiff", "tga",
+                        "gif", "svg"]
+    skip_timelines_check = []  # skip for specific task names (regex)
+
+    def process(self, instance):
+        # Skip the instance if is not active by data on the instance
+        if not self.is_active(instance.data):
+            return
+
+        if any(re.search(pattern, instance.data["task"])
+               for pattern in self.skip_timelines_check):
+            self.log.info("Skipping for {} task".format(instance.data["task"]))
+
+        asset_data = lib.get_asset(instance.data["asset"])["data"]
+        frame_start = asset_data["frameStart"]
+        frame_end = asset_data["frameEnd"]
+        handle_start = asset_data["handleStart"]
+        handle_end = asset_data["handleEnd"]
+        duration = (frame_end - frame_start + 1) + handle_start + handle_end
+
+        repre = instance.data.get("representations", [None])
+        if not repre:
+            self.log.info("No representations, skipping.")
+            return
+
+        ext = repre[0]['ext'].replace(".", '')
+
+        if not ext or ext.lower() not in self.check_extensions:
+            self.log.warning("Cannot check for extension {}".format(ext))
+            return
+
+        files = instance.data.get("representations", [None])[0]["files"]
```

Comment:
This is strange code, it deserve at least some refactor
```suggestion
        repres = instance.data.get("representations")
        if not repres:
            self.log.info("No representations, skipping.")
            return
            
        first_repre = repres[0]
        ext = first_repre['ext'].replace(".", '')

        if not ext or ext.lower() not in self.check_extensions:
            self.log.warning("Cannot check for extension {}".format(ext))
            return

        files = first_repre["files"]
```